sympy/core/add.py

```diff
@@ -345,6 +345,25 @@ def as_coeff_Add(self, rational=False):
     # let Expr.as_coeff_mul() just always return (S.One, self) for an Add.  See
     # issue 5524.
 
+    def _eval_power(self, e):
+        if e is S.Half and self.is_number:
+            from sympy.core.evalf import pure_complex
+            from sympy.functions.elementary.complexes import sign
+            from sympy.functions.elementary.miscellaneous import sqrt
+            n, d = self.as_numer_denom()
```

Comment:
I think I was trying to fail early. I just rewrote this. Let's see how the tests go.